### PR TITLE
feat(Projects): Added new field "domain" to project summary

### PR DESCRIPTION
**Pull request by nutanv1contr**
Summary:
Added new field "domain" to project summary
Field domain" will be a dropdown and the values will be fetched from sw360.properties file
Added rest api support for the field.
Added the field to rest api documentation

Steps to test:
-Create new project and look for a field with name "domain" under project summary.
-The field should of type dropdown and the values should be fetched from sw360.properties file.
-User should be able to select and save the field value.
-User should be able to add the field values by create project from rest end point.
-On entering invalid values rest end point should give error response. 

closes : #539

### CODE REVIEW

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ProjectSpecTest.java [hunk 1]
```diff
@@ -8,6 +8,34 @@
  */
 package org.eclipse.sw360.rest.resourceserver.restdocs;
 
+import static org.eclipse.sw360.datahandler.thrift.MainlineState.MAINLINE;
```
Comment (maxhbr) [inline rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ProjectSpecTest.java]: Resorting the imports looks like an unnecessary refactoring for me. And what IDE do you use, I thin we usually sort the static imports to the bottom of the list and AFAIK this is also the way IntelliJ handles the imports?

**Comment by akapti** [comment]
Tested the changes. working as expected.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectValidator.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import org.eclipse.sw360.datahandler.thrift.projects.Project;
+import org.eclipse.sw360.rest.resourceserver.Sw360ResourceServer;
+import org.springframework.stereotype.Component;
+
+/*
+ * This class is used to validate the project input fields.
+ * @author nutan.das@siemens.com
+ * */
+@Component
+public class ProjectValidator {
```
Comment (imaykay) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectValidator.java]: I like it a lot that you started to use input validation! Do you see any possibility to achieve the same by using more standardized techniques (based on JSRs) like mentioned in https://www.mkyong.com/spring-boot/spring-rest-validation-example/, https://www.baeldung.com/spring-boot-bean-validation or https://www.baeldung.com/spring-data-rest-validators?

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ProjectSpecTest.java (hunk 1 above)
Comment (nutanv1contr) [reply rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ProjectSpecTest.java]: For this case I have reverted the import reordering.

File: frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/basicInfo.jspf [hunk 3]
```diff
@@ -96,6 +97,24 @@
                       placeholder="Enter Description"><sw360:out value="${project.description}" stripNewlines="false"/></textarea>
         </td>
     </tr>
+    <tr><
```
[the remark this answers is not in the document]
Comment (nutanv1contr) [reply frontend/sw360-portlet/src/main/webapp/html/projects/includes/projects/basicInfo.jspf]: Sorry for that.I missed a small typo. Thanks for pointing that out.

**Comment by imaykay** [comment]
Ah, one more thing: could you please double check the styling? It seems that the new row is missing its right border.
image

**Comment by nutanv1contr** [comment]
> Ah, one more thing: could you please double check the styling? It seems that the new row is missing its right border.
> image

**Comment by nutanv1contr** [comment]
> > Ah, one more thing: could you please double check the styling? It seems that the new row is missing its right border.
> > image

Thanks, made necessary changes.This issue seems to be there in chrome.I tested this in fire fox it was working fine.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectValidator.java (hunk 2 above)
Comment (nutanv1contr) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectValidator.java]: Hi @imaykay ,

Currently JSR validation is not working for @BasePathAwareController.(
https://github.com/spring-projects/spring-boot/issues/12863
https://jira.spring.io/browse/DATAREST-593
).
And I don't see any better way to do the validation here.Please suggest if there is any better solution to it.

**Comment by mcjaeger** [comment]
(from discussion) move validation out here and do a clean approach for all of the fields that require validation

**Comment by nutanv1contr** [comment]
Created a separate issue for rest end point field validation. https://github.com/eclipse/sw360/issues/584

**Comment by mcjaeger** [comment]
failure at cvs-search such as also in other branches

**Comment by lepokle** [comment]
@nutanv1contr 
The GUI is implemented in a way, that opening existing projects and saving them will automatically set the first domain value as new domain.

I think it would be better if domain remains unset until it is explicitly selected during project edit.

**Comment by nutanv1contr** [comment]
> @nutanv1contr
> The GUI is implemented in a way, that opening existing projects and saving them will automatically set the first domain value as new domain.
> 
> I think it would be better if domain remains unset until it is explicitly selected during project edit.

Done.

**Pull request merged**